### ISSUE TITLE
Separate data splitting from neural network's training

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#88]
Blocked by #83, PR #85.

**Comment by gcroci2** [comment]
Before actually implementing it, I think it's useful to agree on what we would like to have. The following was my idea:

```python
from deeprank_gnn.NeuralNet import NeuralNet
from deeprank_gnn.DataSet import HDF5DataSet
from deeprank_gnn.ginet import GINet
from deeprank_gnn.models.metrics import OutputExporter, ScatterPlotExporter

database = './hdf5/1ACB_residue.hdf5'

# creating HDF5DataSet object and selecting molecules, node/edge features, target, and clustering method:
dataset = HDF5DataSet(
    root = "./",
    database = database,
    index = range(400),
    node_feature = ["type", "polarity", "bsa", "depth", "hse", "ic", "pssm"],
    edge_feature = ["dist"],
    target = "irmsd",
    clustering_method = "mcl",
)

######### from here what would be new:

# first split between training phase and testing phase, if needed
dataset_train, dataset_test = UDF_DivideDataSet(dataset, percent, shuffle)
# second split for the training phase: training set and validation set 
dataset_train, dataset_valid = UDF_DivideDataSet(dataset_train, percent, shuffle)

# training a network using the above defined dataset
nn = NeuralNet(
    dataset_train, # for the training phase, required
    dataset_valid, # for the training phase, optional
    dataset_test, # for evaluation/testing phase, optional
    GINet,
    task = "reg",
    batch_size = 64
)

nn.train(nepoch = 50)
```

**Comment by cbaakman** [comment]
I think that makes sense.

**Comment by DTRademaker** [comment]
I like the concept, but why not directly "dataset_train, dataset_valid, dataset_test = UDF_DivideDataSet(dataset, percent, shuffle)"?
The advantage of that is that you can use a single seed to do the whole splitting. 
If you need more splits you could for example type: d1, d2, d3, d4, d5 = UDF_DivideDataSet(dataset, percentages=[0.2, 0.2, .3, .1, .2], shuffle)" where the number of outputs is the same as the
 number of percentages.

**Comment by gcroci2** [comment]
> I like the concept, but why not directly "dataset_train, dataset_valid, dataset_test = UDF_DivideDataSet(dataset, percent, shuffle)"? The advantage of that is that you can use a single seed to do the whole splitting. If you need more splits you could for example type: d1, d2, d3, d4, d5 = UDF_DivideDataSet(dataset, percentages=[0.2, 0.2, .3, .1, .2], shuffle)" where the number of outputs is the same as the number of percentages.

Indeed this was my beginning idea, but what if either we don't have or we don't need a testing dataset? But I really like the idea about the percentages. What about this: 

```python
dataset1, dataset2, dataset3 = DivideDataSet(dataset, percent=[perc1, perc2, perc3], shuffle)
```

The number of datasets should correspond to the number of elements in percent list, and I can insert a code check for this. I'll implement the basic case in which we split randomly the dataset in len(percent) datasets. Then the user can implement whatever he/she wants starting from this basic example function.

Then The NeuralNet class will stay the same: 

```python
nn = NeuralNet(
    dataset_train, # for the training phase, required
    dataset_valid, # for the training phase, optional
    dataset_test, # for evaluation/testing phase, optional
    GINet,
    task = "reg",
    batch_size = 64
)
```

**Comment by svenvanderburg** [comment]
I would suggest to not reinvent the wheel, and use the same API as [scikitlearn's `train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html). Of course we can only implement the features that we really need.

But the idea of just passing different train, val, and test sets to `NeuralNet` is very useful!

**Comment by gcroci2** [comment]
Add the possibility to divide the three datasets according to clusters (only distinct values features for now). Also, evaluate the possibility to pass the different clusters' data as list elements to NeuralNet class.

**Comment by gcroci2** [comment]
Blocked by #172
